pkg/runtime/codec.go

```diff
@@ -169,3 +170,27 @@ func (c *parameterCodec) EncodeParameters(obj Object, to unversioned.GroupVersio
 	}
 	return queryparams.Convert(obj)
 }
+
+type base64Serializer struct {
+	Serializer
+}
+
+func NewBase64Serializer(s Serializer) Serializer {
+	return &base64Serializer{s}
+}
+
+func (s base64Serializer) EncodeToStream(obj Object, stream io.Writer, overrides ...unversioned.GroupVersion) error {
+	e := base64.NewEncoder(base64.StdEncoding, stream)
+	err := s.Serializer.EncodeToStream(obj, e, overrides...)
+	e.Close()
```

Comment:
We're closing the base64 encoder, which is required to flush it.  It
doesn't close the stream.

On Thu, May 5, 2016 at 3:25 AM, Wojciech Tyczynski <notifications@github.com
> wrote:

> In pkg/runtime/codec.go
> <https://github.com/kubernetes/kubernetes/pull/24710#discussion_r62155513>
> :
>
> > @@ -169,3 +170,27 @@ func (c *parameterCodec) EncodeParameters(obj Object, to unversioned.GroupVersio
> >  	}
> >  	return queryparams.Convert(obj)
> >  }
> > +
> > +type base64Serializer struct {
> > +	Serializer
> > +}
> > +
> > +func NewBase64Serializer(s Serializer) Serializer {
> > +	return &base64Serializer{s}
> > +}
> > +
> > +func (s base64Serializer) EncodeToStream(obj Object, stream io.Writer, overrides ...unversioned.GroupVersion) error {
> > +	e := base64.NewEncoder(base64.StdEncoding, stream)
> > +	err := s.Serializer.EncodeToStream(obj, e, overrides...)
> > +	e.Close()
>
> hmm - do we really want to close it?
> That would mean that we cannot encode more than one object to the given
> stream, which doesn't sound good to me.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24710/files/e416be61b90b5dc461dc8a84a94d6ecd0d76c216#r62155513>
>